sql/core/src/main/java/org/apache/spark/sql/sources/v2/ReadSupport.java

```diff
@@ -30,6 +30,8 @@
   /**
    * Creates a {@link DataSourceV2Reader} to scan the data from this data source.
    *
+   * If this method fails(throw exception), the query fails before submitting any Spark jobs.
```

Comment:
What is the query you mentioned here? How about?
> If an exception was thrown, we guarantee that no Spark job was submitted.

Reply:
We also need to mention that the "action" will fail...

Reply:
> If an exception was thrown, the action would fail and we guarantee that no Spark job was submitted.